simulator/func_sim/operation.h

```diff
@@ -113,6 +113,15 @@ class Operation
     void set_v_imm( uint32 value) { v_imm = value; }
     auto get_delayed_slots() const { return delayed_slots; }
 
+    Addr get_decoded_target() const
```

Comment:
I suggest to use that method in ALU to calculate `new_PC` to be consinstent.

Reply:
Do you suggest to change this method?
```c++
    void jump( I* instr, Addr target)
    {
        instr->is_taken_branch = true;
        instr->new_PC = target;
        check_halt_trap( instr);
    }
```
Possibly, I can change in such way:
instead of ```instr->new_PC = target;``` we write ```instr->new_PC = tr->get_decoded_target()```.
So, we can remove argument ```Addr target```. But it is wrong because ```jump(I* instr, Addr target)``` is used after for ```indirect jump``` :

```c++
    template<typename I> static
    void jr( I* instr) {
        if (instr->v_src1 % 4 != 0)
            instr->trap = Trap::UNALIGNED_ADDRESS;
        jump( instr, align_up<2>(instr->v_src1));
    }
```
Where we can't throw argument ```target``` as it is known only after EXE stage.

Reply:
What prevents you to call that method in `void j` with different argument?

https://github.com/MIPT-ILab/mipt-mips/blob/1372f0d3a4a28c893e45882b08f0ecce51bf2140/simulator/func_sim/alu.h#L342

Reply:
Ok, I will rewrite it in following way:
```c++
void j( I* instr) { jump(instr, instr->get_decoded_target()); } 
```